django/core/servers/basehttp.py

```diff
@@ -134,6 +134,7 @@ def cleanup_headers(self):
         if (
             self.environ["REQUEST_METHOD"] == "HEAD"
             and "Content-Length" in self.headers
+            and str(self.headers["Content-Length"]) == "0"
```

Comment:
Wouldn't it be technically valid for a 200 response to return a `Content-Length` of `0`?

Reply:
HEAD request Header and subsequent GET request headers should match. So the only time you could have `Content-Length: 0` is if your GET request has an empty body. This is not very likely in general and less likely if you are using a HEAD request since they are most often used in situations where you have a large response.

In a perfect world, there would be a mechanism to see if the user set the content-length. However, checking for 0 is pretty much a good proxy for that without building out a specific mechanism for it. At least that is what I'm thinking.

Reply:
Yeah, given that this is only for runserver,, I think it wouldn't be worth to add an extra mechanism for it.